STM32: Refactor us_ticker files

bcostm opened the pull request:
### Description

This PR re-arrange the files used to manage the us_ticker. Previously the code was mixed in different files (us_ticker.c, hal_tick_16b.c, hal_tick_32b.c, hal_tick_common.c, hal_tick.h). 

Now:
- All the code to manage the us_ticker is in `us_ticker.c `and `us_ticker_data.h` files
- The `hal_tick_overrides.c` file contains only the ST HAL Tick functions redefinitions (weak)

### Pull request type

    [ ] Fix
    [X] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

bcostm committed:
stm32 ticker: rename files and move functions

- rename hal_tick_common.c in hal_tick_overrides.c
- move 16 and 32bits timer functions in us_ticker.c

bcostm committed:
stm32 ticker: rename macro and update ST HAL Tick functions

- rename TIM_MST_16BIT in TIM_MST_BIT_WIDTH in order to use it directly in ticker info structure
- change HAL_InitTick() and HAL_GetTick()

bcostm committed:
stm32 ticker: rename hal_tick.h in us_ticker_data.h

bcostm committed:
stm32 ticker: change license

bcostm committed:
stm32 ticker: typo corrections

cmonr reviewed (approved):
Lot's of cloned file changes. Hoping this was done by script, and not by hand!

Was _much_ easier to tell what happened via each commit instead of the overall diff.

cmonr commented:
/morph build

cmonr commented:
@bcostm Please take a look at the test failures. On quick glance, they all appear to be related to ST boards and tickers.

bcostm commented:
> Lot's of cloned file changes. Hoping this was done by script, and not by hand!

Yes I used sed/find commands to do it (too lazy to do it manually on 55 files :))

I will check the errors

bcostm committed:
stm32 ticker: corrections in order to pass tests

0xc0170 commented:
@bcostm Any update for the failures above?

bcostm commented:
I've managed to make one test pass but it causes other tests to be fail... I put this PR on hold for now until Issue #7316 is fixed because same files/functions are used.

cmonr commented:
For tracking, looks like the PR that will fix #7316 is https://github.com/ARMmbed/mbed-os/pull/7352

cmonr commented:
@bcostm #7352 just came in, so weel free to continue when possible!

bcostm committed:
stm32 ticker: change th eplace where timer init in done, fix overflow issue with 16-bit timer

- Move back the 16/32bit timer initialization in HAL_InitTick() and not in us_ticker_init()
- Use ticker_read_us() and us_ticker_read() in HAL_GetTick() to fix potential overflow issue with the 16bit timer

==> These corrections allow timer, rtc, sleep, tick tests to PASS

bcostm commented:
I have rebased and added what has been done in PR #7352. Tests on several boards on-going. I'll give the status when they'll be terminated.

jeromecoutant commented:
Hi
Results from ST CI bench are all good
Thx

0xc0170 commented:
Can be the commit messages improved? `typo, additional changes, other fixes`  could be: `stm32 ticker: fix macro name` , `stm32 ticker: changing something` , `stm32 ticker: fixing what and why`

bcostm commented:
OK. Commit messages updated.

0xc0170 commented:
/morph build

bcostm commented:
I don't see the `release-version:5.9.3` tag ?
